programs/solana/packages/solana-ibc-macros/src/lib.rs

```diff
@@ -422,3 +429,159 @@ fn generate_discriminator_checks() -> proc_macro2::TokenStream {
         #(#checks)*
     }
 }
+
+// ============================================================================
+// Access Control Macros
+// ============================================================================
+
+/// Adds the access control account field to an Accounts struct.
+///
+/// This macro automatically adds the required `access_manager` field with proper
+/// constraints to your Accounts struct.
+///
+/// # Example
+///
+/// ```ignore
+/// #[derive(Accounts)]
+/// #[with_access_manager]
+/// pub struct RecvPacket<'info> {
+///     // access_manager field is automatically added
+///     #[account(mut)]
+///     pub sender: Signer<'info>,
+///     // ... other accounts
+/// }
+/// ```
+///
+/// Expands to:
+///
+/// ```ignore
+/// #[derive(Accounts)]
+/// pub struct RecvPacket<'info> {
+///     /// Global access control account (owned by access-manager program)
+///     /// CHECK: Validated by seeds constraint pointing to access-manager program
+///     #[account(
+///         seeds = [access_manager::state::AccessControl::SEED],
+///         bump,
+///         seeds::program = access_manager::ID,
+///     )]
+///     pub access_manager: AccountInfo<'info>,
+///
+///     #[account(mut)]
+///     pub sender: Signer<'info>,
+///     // ... other accounts
+/// }
+/// ```
+#[proc_macro_attribute]
+pub fn with_access_manager(_attr: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
ditto not worth it via macros tbh, decreases readability. we're saving few lines of rust code via proc macro
Relying on anchor is also double-edged sword as anchor validations increases function frame size which is not that big in sbf